BridgeSDK/BridgeAPI/SBBActivityManager.m

```diff
@@ -181,9 +180,16 @@ - (NSURLSessionTask *)getScheduledActivitiesForDaysAhead:(NSInteger)daysAhead da
     NSMutableDictionary *headers = [NSMutableDictionary dictionary];
     [self.authManager addAuthHeaderToHeaders:headers];
     
-    // if caching, always request the maximum days ahead from the server so we have them cached
-    NSInteger fetchDaysAhead = gSBBUseCache ? kMaxAdvance : daysAhead;
-    return [self.networkManager get:kSBBActivityAPI headers:headers parameters:@{@"daysAhead": @(fetchDaysAhead), @"offset": [[NSDate date] ISO8601OffsetString]} completion:^(NSURLSessionTask *task, id responseObject, NSError *error) {
+    // If caching, always request the maximum days ahead from the server so we have them cached
+    // and if the desiredDaysAhead is less that the max allowed, then fetch a minimum number of
+    // items per schedule.
+    NSInteger desiredDaysAhead = gSBBUseCache ? gSBBCacheDaysAhead : daysAhead;
+    NSInteger fetchDaysAhead = MIN(kMaxAdvance, desiredDaysAhead);
+    NSInteger fetchMinimumPerSchedule = (fetchDaysAhead < desiredDaysAhead) ? kMaxAdvance : 0;
+    NSDictionary *parameters = @{@"daysAhead": @(fetchDaysAhead),
+                                 @"minimumPerSchedule": @(fetchMinimumPerSchedule),
+                                 @"offset": [[NSDate date] ISO8601OffsetString]};
+    return [self.networkManager get:kSBBActivityAPI headers:headers parameters:parameters completion:^(NSURLSessionTask *task, id responseObject, NSError *error) {
```

Comment:
This is the code that limits the fetch values sent to the server and sets the max to 4 days. If the app is requesting more than 4 days ahead, then the code will fall back to requesting "up to 4 activities per schedule" (for those schedules that do not return any activities within the next 4 days.)

For example:

In Smart4SURE, all schedules are either 1 week and 12 weeks. So they will all return 4 activities per schedule. This is then filtered by the requested daysAhead and daysBehind before returning in the completion handler. (In that case, with a 10 day window, it will always filter to no more than 1 schedule per activity)

In mPower, where activities are scheduled 3x daily, the server will return no more than 4 days out for those schedules (or 12 activities) but will will still return a monthly activity which could be displayed in the "coming up" section of the activities view.